src/autocat/saa.py

```diff
@@ -34,6 +34,21 @@ def _find_dopant_index(structure, dopant_element):
     return dopant_index[0][0]
 
 
+def _find_all_surface_atom_indices(structure, tol=None) -> List[int]:
```

Comment:
I'd recommend `def _find_all_surface_atom_indices(structure, tol: float = 0.5) -> List[int]:`, and removing the `if tol is None` as that format isn't needed for immutable argument types like float.